### ISSUE TITLE
Started to get errors on RevokeTokenAsync since moving up to 4.1.1 

### ISSUE THREAD

**Issue by leealexander** [DuendeArchive/IdentityModel#285]
Since moving up from IdentityModel 3.10.6 to 4.1.1 we have been getting errors when calling RevokeTokenAsync using ADFS as the IDP. 

Having a look at the 4.1.1 code for RevokeTokenAsync it makes the request and then passes the response to ProtocolResponse.FromHttpResponseAsync. The response status code is 200 and the content returned is HTML. The code however tries to read the content as json and then fails with an exception which is caught and placed into ProtocolResponse.Exception with IsError set to true.

In the [rfc7009 ](https://tools.ietf.org/html/rfc7009)it says 

>    The authorization server responds with HTTP status code 200 if the
>    token has been revoked successfully or if the client submitted an
>    invalid token.
> 
>    Note: invalid tokens do not cause an error response since the client
>    cannot handle such an error in a reasonable way.  Moreover, the
>    purpose of the revocation request, invalidating the particular token,
>    is already achieved.
> 
>    The content of the response body is ignored by the client as all
>    necessary information is conveyed in the response code.

If there is an error response 400 then it conforms to the definition in [Section 5.2 of [RFC6749]](https://tools.ietf.org/html/rfc6749#section-5.2)

So, I’m wondering why the code is trying to parse the response given that the result is 200? In the old code 3.10.6, if a 200 was returned it would return a new TokenRevocationResponse which by default was a non-error.

My workaround for now is to ignore the error if  ErrorType !=ResponseErrorType.Protocol.

Regards
Lee

**Comment by leastprivilege** [comment]
Why is there HTML coming back?

**Comment by leealexander** [comment]
I don't know why the HTML is coming back, the html is:
"Sign out
You have signed out or your session has been idle for too long. If you're signing out please close your browser."

The IDP is ADFS although we develop and test (to a point) using IdentityServer. It makes life much easier - then we test and release using ADFS which is the live system which we have no control over. It's the ADFS IDP that sends back HTML.

We revoke all the tokens one by one calling RevokeTokenAsync with the appropriate token hint, then we call HttpContext.SignOutAsync.

**Comment by leastprivilege** [comment]
I didn’t know that Adfs  supports revocation. are you sure it does?

**Comment by leealexander** [comment]
I think you are correct! When we switched to ADFS from solely using IdentityServer whilst we were developing the product, one of the differences we hit was the [lack of a TokenIssuer property](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1030) on the discovery document. 

This was also the case with the RevocationEndpoint property; I assumed it was a similar problem and since the  3.10.6 version of IdentityModel RevokeTokenAsync  seemed to work with the value I pulled out of the end_session_endpoint property I assumed all was good.

I didn't realise the revocation of tokens was an optional thing. 

Thanks for your help.
